src/components/NewContainerSearch.react.js

```diff
@@ -184,6 +197,32 @@ module.exports = React.createClass({
           {otherResults}
         </div>
       );
+    } else if (filter === 'userimages') {
```

Comment:
Can you move this above the `else if (this.state.loading)` clause? It will make sure loading doesn't appear if they type in the search bar while browsing `My Images`.